### PR TITLE
Fix file upload bug.

### PR DESCRIPTION
**Pull request by RobertCram**
PT Story: https://www.pivotaltracker.com/story/show/153337811

**Changes proposed in this pull request:**

1. Clear the "files to upload" list every time the file upload button is used, so it is clear to the user that only the last selected files will be uploaded.
 
2. Add a feature test for the situation where the upload button is used multiple times.

Ready for review:
@weedySeaDragon @patmbolger @thesuss

### CODE REVIEW

**Review by thesuss — APPROVED** [review]

File: app/views/membership_applications/_membership_application_fields.html.haml [hunk 1]
```diff
@@ -78,7 +78,8 @@
     var upload_table = document.getElementById("files-to-upload");
 
     var filesUploadButton = document.getElementById("upload-button");
-    var txt = "";
+
+    $(upload_table).find("tr>td").remove();
```
Comment (weedySeaDragon) [inline app/views/membership_applications/_membership_application_fields.html.haml:80]: I think a user should be able to _add_ to the list of uploaded files.

But for now, if we don't want to handle that, we should at least **warn the user** that their current list of files will be wiped out.

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]

File: app/views/membership_applications/_membership_application_fields.html.haml (hunk 1 above)
Comment (patmbolger) [inline app/views/membership_applications/_membership_application_fields.html.haml:80]: The user _can_ upload multiple files at once if they select multiple files in a single file-select dialog box.

Perhaps we should add a tooltip that lets them know they can do this (so if they _do_ want to upload multiple files it is less frustrating - although we also need to fix this eventually so they can upload multiple files by clicking the "Choose files ..." button multiple times).

Here is the form when multiple files are specified in the dialog box:

screen shot 2017-12-04 at 6 23 23 pm

And here is the result of clicking the "save" button:

screen shot 2017-12-04 at 6 23 58 pm

Comment (thesuss) [reply app/views/membership_applications/_membership_application_fields.html.haml:80]: Good idea with a Tooltip, can we have it in and shown at the client meeting and then perhaps deployed after?

Comment (patmbolger) [reply app/views/membership_applications/_membership_application_fields.html.haml:80]: The tooltip (or some other "help" mechanism) should explain how to select multiple files on a Mac and on windows.

Eventually, we could redesign the file upload (sub)form to allow the user to add another file to upload (as many times as they want), using, perhaps, the cocoon gem (which works very well for this kind of situation).  (https://github.com/nathanvda/cocoon)

Comment (patmbolger) [reply app/views/membership_applications/_membership_application_fields.html.haml:80]: @weedySeaDragon - see below the suggestion to add some help text explaining how to upload multiple files.  I think that negates the need for a warning about wiping out previously-selected files (and adding both the help text for multiple selection _and_ the warning text could be very confusing, IMO).

Comment (weedySeaDragon) [reply app/views/membership_applications/_membership_application_fields.html.haml:80]: What about doing some simple text like this for now:  
- Add text that tells them they can upload multiple files at once (_"You can upload multiple files..."_  in my rough example below)
- Add explicit instructions for Mac and PC below that.

---
multi-file upload instructions idea


---
That way it is not hidden in a tooltip.  
If people using this are not power-users, then they need instructions to stay visible so they can follow the steps as they do something.  When it's in a pop-up tooltip, they can't keep referring to it easily.  (Pop-up tootips are good for quick reminders)

(I am assuming that most members are _not_ "power users".)

Comment (patmbolger) [reply app/views/membership_applications/_membership_application_fields.html.haml:80]: I like this idea a lot better than a tooltip :)

File: app/views/membership_applications/_membership_application_fields.html.haml [hunk 2]
```diff
@@ -78,7 +78,8 @@
     var upload_table = document.getElementById("files-to-upload");
 
     var filesUploadButton = document.getElementById("upload-button");
-    var txt = "";
+
```
Comment (patmbolger) [inline app/views/membership_applications/_membership_application_fields.html.haml:79]: BTW, on line 73 there is this statement, which should be removed:
```
console.log('upload-button change');
```

**Review by weedySeaDragon — APPROVED** [review]

**Review by patmbolger — APPROVED** [review]

**Comment by patmbolger** [comment]
As agreed at yesterday's team meeting, we decided to defer the enhancement for an improved UX to another story.

Merging this now ....

**Pull request merged**